src/main/java/org/commcare/formplayer/services/EntitiesSelectionService.java

```diff
@@ -0,0 +1,72 @@
+package org.commcare.formplayer.services;
+
+import com.timgroup.statsd.StatsDClient;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.commcare.core.interfaces.EntitiesSelectionCache;
+import org.commcare.formplayer.objects.EntitiesSelection;
+import org.commcare.formplayer.repo.EntitiesSelectionRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.CacheConfig;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.annotation.Nullable;
+
+@Service
+@CacheConfig(cacheNames = {"entities_selection"})
+public class EntitiesSelectionService implements EntitiesSelectionCache {
+
+    private final Log log = LogFactory.getLog(EntitiesSelectionService.class);
+
+    @Autowired
+    private EntitiesSelectionRepo entitiesSelectionRepo;
+
+    @Autowired
+    private FormplayerStorageFactory storageFactory;
+
+    @Autowired(required = false)
+    private StatsDClient datadogStatsDClient;
+
+    @Autowired
+    CacheManager cacheManager;
+
+    @Override
+    public UUID write(String[] values) {
+        EntitiesSelection entitySelection =  new EntitiesSelection(storageFactory.getUsername(),
```

Comment:
I am a bit unsure if `storageFactory` is the right object to take the metadata from and have not been able to test this yet. Though hoping to do it soon.

Reply:
I actually think the restore factory is a better choice at the moment.

Reply:
Ahh ok, curious if you see anything wrong with `storageFactory` ? It seems like it exists to provide this data and seems to be more light weight than `restoreFactory`.

Reply:
I think it's fine to use either provided it is available in all the requests where it's needed (which it seems like it should be). Ultimately I want to extract this out into the auth objects.

Reply:
Sounds good!